R/calcEmiTarget.R

```diff
@@ -49,18 +67,71 @@ calcEmiTarget <- function(sources, subtype, scenario) {
     )
   }
 
+  # convert target factors to magclass objects
   # ensure that all magclass objects in the list have matching years so they can be bound together
-
   listYears <- lapply(listGhgFactors, getItems, dim = "year") %>% unlist() %>% unique() %>% sort()
   ghgFactor <- purrr::map(listGhgFactors,
                           ~ add_columns(.x, listYears[!listYears %in% getItems(.x, dim = "year")], 2))
-
   ghgFactor <- mbind(ghgFactor)
   ghgFactor <- ghgFactor[, sort(getYears(ghgFactor)), ]
 
-  # GHG factors weighted by GHG in 2015
-  if (subtype == "Ghgfactor") {
 
+  # 2. Extrapolate NDC targets from 2030 to 2035 for countries which do not have 2035 NDC targets ----
+
+  # add NDC versions "Emissions_NDC " and "Emissions_2025_uncond_extrapol"
+  # that take the 2035 targets from the 2025 version, but extrapolate 2030 targets to 2035 for countries which do not have 2035 targets yet
+  # create new NDC version 2025 with extrapolated targets
+
+  # if no 2035 targets available, add 2035 to target factor object
+  if (! "y2035" %in% getYears(ghgFactor)) {
+    ghgFactor <- mbind(ghgFactor, new.magpie(getRegions(ghgFactor),"y2035",getNames(ghgFactor), fill = NA))
+  }
+  ghgFactorExtrapolated <- ghgFactor[,,c("2025_cond","2025_uncond")]
+  getItems(ghgFactorExtrapolated, dim=3.1) <- paste0(getItems(ghgFactorExtrapolated, dim=3.1),"_extrapol")
+  # loop over countries and NDC scenarios to extrapolate 2030 targets to 2035 targets for those countries who do not have 2035 targets
+  for (country in getRegions(ghgFactorExtrapolated ) ) {
+    for (NDCscenario in getItems(ghgFactorExtrapolated, dim=3) ) {
+      # do extrapolation only if country has 2030 target but no 2035 target
+      if (c("y2035") %in% getYears(ghgFactorExtrapolated)) {
```

Comment:
This condition should always be true due to line 86-88

Btw: you can also do integer comparison: 
`if (2035 %in% getYears(ghgFactorExtrapolated, as.integer = TRUE)) {`